docfx_project/articles/bootstrapping-trust/part-03.certificates.md

```diff
@@ -0,0 +1,180 @@
+certificate is invalid and should not be trusted. Validity periods
+should be as small as possible to fit the use case. Shorter periods
+reduce the window of time that compromised private key can remain useful
+for an attack. The cost of this is overhead reissuing certificates as
+they reach the end of their validity period.
+
+#### Usage/Extensions
+
+Usage/Extension Data is interesting because it can limit what roles a
+certificate fulfills. Depending on the system, this may be adhered to or
+not. Some examples of usage that are common to see:
+
+- key usage: client authentication, server authentication, signatures,
+  etc.
+- Subject Alternate Names (SANs)
+  - Limits what IP address, email address, domain name, etc. the
+    certificate can be associated with
+- Certificate Authority (CA) flag
+- and more
+
+This series will not dive into the details of these usages. However, it
+is essential to be aware of them and that they can affect the roles a
+certificate can play and that it is necessary to configure them
+correctly.
+
+#### Signatures
+
+The signature section of a certificate is a list of signatures from
+other entities that trust this certificate. A certificate that signs
+itself is a "self-signed certificate." Self-signed certificates must be
+individually trusted as no other certificate has expressed trust in it
+by signing it. Self-signed certificates sometimes used for testing
+purposes as they have a low barrier to entry for creating a certificate.
+They are also used as Root Certificate Authorities (root CAs).
+
+Each signature is the result of taking the contents of the certificate
+(without signatures), one-way hashing it, and then encrypting the hash
+with the signator's private key. The result is appended to the end of
+the signature list. During this process, the public certificate moves
+between systems to be signed.
+
+The movement of the public certificate between systems is facilitated by
+Certificate Signing Requests (CSRs). CSRs can be transmitted
+electronically as files or as a data stream to the signer. CSRs contain
+only the public information of a certificate and a signature from the
+certificate's private key holder. Since they only contain public
+information, they are not considered sensitive. The signature allows the
+signer to verify that the CSR is from the subject specified in the CSR.
+If the signature is valid, the signator processes the CSR, and the
+result is a newly minted certificate with an additional signature.
+
+### Conclusion
+
+Certificates are keys, usually public ones, with additional metadata
+that adds conventions and restrictions around certificate usages. They
+provide a place for signatures to resides and, through CSRs, provide a
+vehicle to request additional signatures. Certificates are useful
+because they package all of these concerns into a neat single file. In
+the next part, we will explore how to create a formidable chain of
```

Comment:
In the next part -> "In [part 3](./part3.md)"

Reply:
fixed.

Reply:
Also added on from p1 to p2 and p2 to p3.